src/main/java/org/apache/commons/fileupload/FileUploadBase.java

```diff
@@ -206,6 +206,18 @@ public long getSizeMax() {
         return sizeMax;
     }
 
+    /**
+     * Gets allowed max size for current upload, subclasses can override this method
+     * to calculate allowed size per request
+     * @param ctx requestContext - provides information related to current upload request.
+     * @return max size allowed for current request
+     */
+    public long getSizeMaxForThisRequest(RequestContext ctx) {
+        return sizeMax;
```

Comment:
What is the difference between this method implementation and the `getSizeMax` method?

The `ctx` argument passed in is also not used.

Reply:
ctx is to be used in subclasses of FileUploadBase, so that max allowed size can be calculated based on request attributes at runtime. It's an extension points to allow dynamic upload limit.